api/src/main/java/org/hawkular/inventory/api/filters/RelationWith.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.inventory.api.filters;
+
+import org.hawkular.inventory.api.model.Entity;
+
+import java.util.Arrays;
+
+/**
+ * @author Jirka Kremser
+ * @since 1.0
+ */
+public final class RelationWith {
+    private RelationWith() {
+
+    }
+
+    public static Ids id(String id) {
+        return new Ids(id);
+    }
+
+    public static Ids ids(String... ids) {
+        return new Ids(ids);
+    }
+
+    public static Properties property(String property, String value) {
```

Comment:
Good idea, but let's not complicate it. This ought to be about the `Relationship service`, right? I am happy to add it afterwards.